### ISSUE TITLE
Application is both mixing in the config into itself and to a params field

### ISSUE THREAD

**Issue by cjolif** [dmachi/dojox_application#65]
See:

``` javascript
constructor: function(params, node){
  lang.mixin(this, params);
  this.params = params;
  // ...
}

```

This seems redundant. I think only params should be kept the mixin is dangerous. For example Application defines a controller object that contains the instances of controllers and the config also contains such a property.

**Comment by cjolif** [comment]
View is doing only mixin of params. I would say at least we should be consistent between the app and view. So either only mixin in both or only set a params field (preferred in my opinion as it prevents any potential clash).

**Comment by dmachi** [comment]
I agree that it should probably be one or the other and don't feel overly strong about either way.   On the one hand mixing in  can have clashes as cjolif points out, but on the other hand the whole point of that object is to make it easy to config/change/override an app in different situations.  All that said, we can use params and then set properties on the main instances by inspecting that object, so its an extra step but no loss of capabilities.
